sdk/storage/azure-storage-file-share/tests/test_nfs_async.py

```diff
@@ -0,0 +1,262 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import pytest
+from azure.core.exceptions import ClientAuthenticationError, HttpResponseError, ResourceExistsError, ResourceNotFoundError
+from azure.storage.fileshare import ContentSettings, ShareServiceClient
+from azure.storage.fileshare.aio import ShareServiceClient as AsyncShareServiceClient
+from azure.storage.fileshare.aio import ShareFileClient, ShareDirectoryClient
+
+from devtools_testutils.aio import recorded_by_proxy_async
+from devtools_testutils.storage.aio import AsyncStorageRecordedTestCase
+from settings.testcase import FileSharePreparer
+
+
+TEST_INTENT = 'backup'
+TEST_FILE_PREFIX = 'file'
+TEST_DIRECTORY_PREFIX = 'directory'
+
+
+class TestStorageFileNFSAsync(AsyncStorageRecordedTestCase):
+
+    fsc: ShareServiceClient = None
+
+    async def _setup(self, storage_account_name):
+        self.account_url = self.account_url(storage_account_name, 'file')
+        self.credential = self.get_credential(AsyncShareServiceClient, is_async=True)
+        self.fsc = AsyncShareServiceClient(
+            account_url=self.account_url,
+            credential=self.credential,
+            token_intent=TEST_INTENT
+        )
+        self.share_name = self.get_resource_name('utshare')
+
+        async with AsyncShareServiceClient(
+            account_url=self.account_url,
+            credential=self.credential,
+            token_intent=TEST_INTENT
+        ) as fsc:
+            if self.is_live:
+                try:
+                    await fsc.create_share(self.share_name, protocols='NFS')
+                except:
+                    pass
+
+    def teardown_method(self):
+        fsc = ShareServiceClient(
+            account_url=self.account_url,
+            credential=self.get_credential(ShareServiceClient),
+            token_intent=TEST_INTENT
+        )
+        if fsc:
```

Comment:
This should be `self.fsc`. You want to check that the async class one was created which means the `_setup` method was run. Also you'll want to move the creation of the sync client under the if as well. So:

```py
if self.fsc:
    fsc = ShareServiceClient(
        account_url=self.account_url,
        credential=self.get_credential(ShareServiceClient),
        token_intent=TEST_INTENT
    )
   ...
```

Reply:
Got it, this was a typo 😅 . It wouldn't make much sense to check `fsc` because we just created it :)